src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -1221,6 +1280,27 @@ private bool ScanNumericLiteral(ref TokenInfo info)
             return true;
         }
 
+        // TODO: Change to Int64.TryParse when it supports NumberStyles.AllowBinarySpecifier (inline this method into GetValueUInt32/64)
+        private bool TryParseBinaryUInt64(string text, out ulong value)
+        {
+            value = 0;
+            for (int i = 0; i < text.Length; i++)
```

Comment:
@gafter Slightly more efficient the way @khyperia has written since there no enumerator